hack/tools/kapinger/clients/dns.go

```diff
@@ -43,3 +44,16 @@ func (k *KapingerDNSClient) MakeRequests(ctx context.Context) error {
 		}
 	}
 }
+
+func randomString(n int) string {
+	var letters = []rune("abcdefghijklmnopqrstuvwxyz")
+	b := make([]rune, n)
+
+	rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
Does the New function return a type? On Line 55 I would expect that to use the instance. 

r := rand.New(.....)

on line 55 ...

r.IntN(......)

I don't think this is doing what you think it is. Perhaps create a unit test for this function and verify.

Reply:
It will kind of work :) `rand.Intn` uses a package global logger, which I believe is initialized to the same seed. The new random source generated on this line is discarded, which is likely not intended.